dbt_platform_helper/utils/aws.py

```diff
@@ -349,6 +350,46 @@ def get_postgres_connection_data_updated_with_master_secret(session, parameter_n
     return parameter_data
 
 
+def get_redis_supported_versions(elasticache_client=None):
+
+    supported_versions = []
+
+    if not elasticache_client:
+        session = get_aws_session_or_abort()
+        elasticache_client = session.client("elasticache")
+
+    supported_versions_response = elasticache_client.describe_cache_engine_versions(Engine="redis")
+
+    supported_versions = [
+        version["EngineVersion"] for version in supported_versions_response["CacheEngineVersions"]
+    ]
+
+    write_to_cache("redis", supported_versions)
+
+    return supported_versions
+
+
+def get_opensearch_supported_versions(opensearch_client=None):
```

Comment:
Reads a little better...
```suggestion
def get_supported_opensearch_versions(opensearch_client=None):
```